src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -884,6 +884,11 @@ private TypeSymbol BindVariableType(CSharpSyntaxNode declarationNode, Diagnostic
                         }
                     }
                 }
+                else if (initializerOpt is BoundStackAllocArrayCreation boundStackAlloc)
+                {
+                    declTypeOpt = new PointerTypeSymbol(boundStackAlloc.ElementType);
+                    initializerOpt = GenerateConversionForAssignment(declTypeOpt, boundStackAlloc, diagnostics, false, localSymbol.RefKind);
```

Comment:
>localSymbol.RefKind); [](start = 119, length = 21)

Do we have a unit test that ensures we fail if you try and use a ref Span<T> local? 

``` c#
ref Span<int> s = stackalloc int[5]
```